crates/runtime/src/lib.rs

```diff
@@ -63,6 +64,47 @@ pub use crate::vmcontext::{
     VMSharedSignatureIndex, VMTableDefinition, VMTableImport, VMTrampoline, ValRaw,
 };
 
+mod module_id;
+pub use module_id::{CompiledModuleId, CompiledModuleIdAllocator};
+
+#[cfg(feature = "memfd-allocator")]
+mod memfd;
+
+/// When memfd support is not included, provide a shim type and
+/// constructor instead so that higher-level code does not need
+/// feature-conditional compilation.
+#[cfg(not(feature = "memfd-allocator"))]
+#[allow(dead_code)]
+mod memfd {
+    use anyhow::Result;
+    use std::sync::Arc;
+    use wasmtime_environ::{DefinedMemoryIndex, Module};
+
+    /// A shim for the memfd image container when memfd support is not
+    /// included.
+    pub enum ModuleMemFds {}
+
+    /// A shim for an individual memory image.
+    #[allow(dead_code)]
+    pub enum MemoryMemFd {}
+
+    impl ModuleMemFds {
+        /// Construct a new set of memfd images. This variant is used
+        /// when memfd support is not included; it always returns no
+        /// images.
+        pub fn new(_: &Module, _: &[u8]) -> Result<Option<Arc<ModuleMemFds>>> {
+            Ok(None)
+        }
+
+        /// Get the memfd image for a particular memory.
+        pub(crate) fn get_memory_image(&self, _: DefinedMemoryIndex) -> Option<&Arc<MemoryMemFd>> {
+            None
```

Comment:
I think where possible it's best to replace this with something like `match self {}` which the compiler will allow since `self` is an empty enum and is a good way of showing that this is statically not reachable.

Reply:
Done!